tests/test_pytesseract.py

```diff
@@ -1,25 +1,246 @@
+# encoding: utf-8
+
 import os.path
+from multiprocessing import Pool
+from sys import platform, version_info
 
 import pytest
+from pytesseract import (
+    Output,
+    TSVNotSupported,
+    get_tesseract_version,
+    image_to_boxes,
+    image_to_data,
+    image_to_osd,
+    image_to_pdf_or_hocr,
+    image_to_string
+)
+from pytesseract.pytesseract import prepare
+
+try:
+    from PIL import Image
+except ImportError:
+    import Image
+
+try:
+    import pandas
+except ImportError:
+    pandas = None
+
+
+IS_PYTHON_2 = version_info[:1] < (3, )
+IS_PYTHON_3 = not IS_PYTHON_2
 
-from pytesseract import image_to_string
+TESSERACT_VERSION = tuple(get_tesseract_version().version)  # to skip tests
 
 DATA_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
 
 
-@pytest.mark.pytesseract
-def test_quick_brown_dog_image():
-    test_file = os.path.join(DATA_DIR, 'test.png')
+pytestmark = pytest.mark.pytesseract  # used marker for the module
 
+
+@pytest.fixture(scope='session')
+def test_file():
+    return os.path.join(DATA_DIR, 'test.png')
+
+
+@pytest.mark.parametrize('test_file', [
+    # https://github.com/tesseract-ocr/tesseract/issues/2558
```

Comment:
Can these tests be turned on for tesseract 4.0 / bionic?

Reply:
It also depends on the behaviour and version of Leptonica. On my system I have the newest versions of Tesseract and Leptonica and it doesn't work. Nevertheless I will test it with the images of Travis.

JFYI, we can skip tests with this decorator:

```python
@pytest.mark.skipif(
    TESSERACT_VERSION[0] < 4,
    reason='requires tesseract >= 4'
)
# ...
```